better_errors.gemspec

```diff
@@ -33,4 +33,10 @@ Gem::Specification.new do |s|
   # optional dependencies:
   # s.add_dependency "binding_of_caller"
   # s.add_dependency "pry"
+  
+  if s.respond_to?(:metadata)
```

Comment:
I’d suggest removing this condition, otherwise anyone building this gem on an older version of RubyGems will build a gem without metadata without being aware of it.

Reply:
That's on purpose really. I'm assuming that the release managers for this repo have the recent rubygems and will publish the metadata, but on the other hand, the check prevents older rubygems from crashing on build.

This is also how rubygems serialises this file:

```
[8] pry(main)> s=Gem::Specification.new('foo')
=> Gem::Specification.new do |s| s.name = "foo" s.version = nil s.installed_by_version = Gem::Version.new("0") s.date = Time.utc(2018, 9, 3) s.require_paths = ["lib"] s.rubygems_version = "2.7.6" s.specification_version = 4 s.summary = nil end
[9] pry(main)> s.metadata['foo'] = 'bar'
=> "bar"
"end\n"
[10] pry(main)> puts s.to_ruby
...
  s.required_rubygems_version = Gem::Requirement.new(">= 0".freeze) if s.respond_to? :required_rubygems_version=
  s.metadata = { "foo" => "bar" } if s.respond_to? :metadata=
  s.require_paths = ["lib".freeze]
...
=> nil
```

Let me know if you'd still like to drop this condition.

Reply:
Let's push this forward. I added a warning in case the metadata is not supported.